### PR TITLE
feat: Adds useInterval

### PR DESCRIPTION
**Pull request by edas**
Allows to rerender the calling component by a given interval

### CODE REVIEW

**Comment by ptbrowne** [comment]
No tests :) ?

**Comment by edas** [comment]
> No tests :) ?

Fixed, thank  you

File: react/hooks/useInterval.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * Force a new render each `duration` ms
+ *
+ * @param {number} duration - in millisecond
+ * @returns {Date} last trigger date for a rerender
+ */
+export default function useInterval(duration) {
```
Comment (y-lohse) [inline react/hooks/useInterval.js]: Have you considered a name like `setIntervalRerender` or something similar? To make it explicit that the hook is about re-rendering and not so much about intervals.

File: react/hooks/useInterval.spec.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import useInterval from './useInterval'
+import { renderHook } from '@testing-library/react-hooks'
+
+async function wait(duration) {
+  console.log('setTimeout', duration)
```
Comment (y-lohse) [inline react/hooks/useInterval.spec.js]: We probably don't want the extra log here :)

File: react/hooks/useInterval.spec.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  return new Promise(resolve => {
+    window.setTimeout(() => {
```
Comment (y-lohse) [inline react/hooks/useInterval.spec.js]: Could we use [timer mocks here](https://jestjs.io/docs/en/timer-mocks.html)?

File: react/hooks/useInterval.js (hunk 1 above)
Comment (ptbrowne) [reply react/hooks/useInterval.js]: usePeriodicRender

File: react/hooks/useInterval.spec.js (hunk 3 above)
Comment (edas) [reply react/hooks/useInterval.spec.js]: Vu ensemble, pas adapté pour ce cas

**Pull request merged**